app/Console/ProcessDuplicatePlaces.php

```diff
@@ -0,0 +1,157 @@
+
+    private AMQPPublisher $amqpPublisher;
+
+    private DocumentEventFactory $eventFactoryForEvents;
+
+    private Connection $connection;
+
+    public function __construct(
+        CommandBus $commandBus,
+        DuplicatePlaceRepository $duplicatePlaceRepository,
+        CanonicalService $canonicalService,
+        AMQPPublisher $amqpPublisher,
+        DocumentEventFactory $eventFactoryForEvents,
+        RepositoryInterface $eventRelationsRepository,
+        Connection $connection
+    ) {
+        $this->duplicatePlaceRepository = $duplicatePlaceRepository;
+        $this->canonicalService = $canonicalService;
+        $this->amqpPublisher = $amqpPublisher;
+        $this->eventFactoryForEvents = $eventFactoryForEvents;
+        $this->eventRelationsRepository = $eventRelationsRepository;
+        $this->connection = $connection;
+
+        parent::__construct($commandBus);
+    }
+
+    public function configure(): void
+    {
+        $this->setName('place:process-duplicates');
+        $this->setDescription('Process duplicate places (determine canonical, update event locations and reindex)');
+        $this->addOption(
+            'dry-run',
+            'd',
+            InputOption::VALUE_NONE,
+            'Execute a dry-run of the process-duplicates script.'
+        );
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $dryRun = (bool)$input->getOption('dry-run');
+
+        $clusterIds = $this->duplicatePlaceRepository->getClusterIds();
+
+        if (count($clusterIds) === 0) {
+            $output->writeln('No clusters found to process');
+            return 0;
+        }
+
+        if (!$this->askConfirmation($input, $output, count($clusterIds))) {
+            return 0;
+        }
+
+        foreach ($clusterIds as $clusterId) {
+            // 1. Set the canonical of a cluster
+            try {
+                $canonicalId = $this->canonicalService->getCanonical($clusterId);
+            } catch (MuseumPassNotUniqueInCluster $museumPassNotUniqueInClusterException) {
+                $output->writeln($museumPassNotUniqueInClusterException->getMessage());
+                continue;
+            }
+
+            $output->writeln('Setting ' . $canonicalId . ' as canonical of cluster ' . $clusterId);
+            if (!$dryRun) {
+                $this->duplicatePlaceRepository->setCanonicalOnCluster($clusterId, $canonicalId);
+            }
+
+            // 2. Trigger a SAPI3 reindex on the places in duplicate_places and places removed from duplicate_places
+            $placesToReIndex = array_merge(
+                $this->duplicatePlaceRepository->getPlacesInCluster($clusterId),
+                $this->getDuplicatePlacesRemovedFromCluster()
+            );
+            foreach ($placesToReIndex as $placeToReIndex) {
+                $placeProjected = $this->eventFactoryForEvents->createEvent($placeToReIndex);
```

Comment:
Shouldn't this use the "event factory for places" instead? Since this is a place "projection update"?